libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -935,30 +940,41 @@ namespace meshkernelapi
                 throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
             }
 
-            if (!meshKernelState.contains(meshKernelId))
+            if (!meshKernelState[meshKernelId].m_curvilinearGrid->IsValid())
             {
-                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+                throw meshkernel::MeshKernelError("Invalid curvilinear grid");
             }
 
-            if (!meshKernelState[meshKernelId].m_curvilinearGrid->IsValid())
+            if (boundaryPolygons.coordinates_x == nullptr || boundaryPolygons.coordinates_y == nullptr)
             {
-                throw meshkernel::MeshKernelError("Invalid curvilinear grid");
+                throw meshkernel::MeshKernelError("Boundary polygon array are null");
             }
 
-            auto lowerLeftNUnsigned = static_cast<meshkernel::UInt>(lowerLeftN);
-            auto lowerLeftMUnsigned = static_cast<meshkernel::UInt>(lowerLeftM);
-            auto upperRightNUnsigned = static_cast<meshkernel::UInt>(upperRightN);
-            auto upperRightMUnsigned = static_cast<meshkernel::UInt>(upperRightM);
+            if (meshKernelState[meshKernelId].m_boundariesAsPolygonCache == nullptr)
+            {
+                throw meshkernel::MeshKernelError("Polygon data has not been cached");
```

Comment:
throw meshkernel::MeshKernelError("Polygon data has not been cached, mkernel_curvilinear_count_boundaries_as_polygons must be called before");? but we will be facing this scenario as developers, because users will use the the wrapped meshkernel api